modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptFetchClientCodegen.java

```diff
@@ -134,14 +134,15 @@ public void processOpts() {
 
     @Override
     public String getTypeDeclaration(Schema p) {
+        String oasType = getSchemaType(p);
         if (ModelUtils.isFileSchema(p)) {
             return "Blob";
         } else if (ModelUtils.isBinarySchema(p)) {
             return "Blob";
         } else if (ModelUtils.isDateSchema(p)) {
-            return "Date";
+            return typeMapping.containsKey("Date") ? typeMapping.get("Date") : "Date";
         } else if (ModelUtils.isDateTimeSchema(p)) {
-            return "Date";
+            return typeMapping.containsKey("Date") ? typeMapping.get("Date") : "Date";
         }
```

Comment:
I don't think there is a need to have custom type mapping for datetime here at all - because AbstractTsClient already includes it into its `typeMappings` map. That's why it's not a problem for the TsAngularClient, which doesn't define any specific mapping for `DateTime` at all, and it still works.
So we could just remove it from the child TS generators and rely on the abstract one

Any thoughts?

Reply:
The custom mapping was apparently introduced in 1e023f3b6e4d655d6e553bb359ffb128f5a20382 to fix a bug where someone expected the default type for date to be `Date`.

I agree that it seems a bit weird and that changing the defaults is exactly what type-mappings is for. Is it reasonable to change the (now) default behavior for date back to `string`?

Reply:
There are a few things going on here:
- Abstract TS client has default mapping for date and DateTime (latter maps to typescript Date object - by mistake)
- TsFetch client doesn't define its own mappings, but redefines getTypeDeclaration in a way that it's always Date.
- TsFetch client also defines some runtime logic in the generated TS code to actually instantiate Date objects from the strings. So does TsReduxQuery.
- TsRxjs client doesn't generate any mapping code, so `Date` type in the interfaces is simply incorrect there again.

Based on the above, TsFetch's usage of Date object is legit - because the generated code indeed converts raw http data to and from Date. (the same also true for TsReduxQuery).
The only reason for a user to change this default behavior is if they have custom templates (proivided via `--template-dir`), which do not do any Date serialization, in contrast to built-in templates.

I can imagine the following solution:
- All child TS clients rely on AbstractTs's protected typeMappings. Furthermore, we want to make sure it maps `DateTime` to string, as it already does for `date`.
- **Some** child TS clients (fetch, redux-query) should override those typeMappings - not via custom Java code, but via the same protected `typeMappings`.
- If a user has special needs (because they use custom mustache templates), they provide custom mappings typeMappings via `--type-mappings` option, which takes precedence over values populated in the codegen constructor.

Reply:
Right, that sounds sensible.